app/controllers/katello/api/v2/subscriptions_controller.rb

```diff
@@ -3,7 +3,7 @@ class Api::V2::SubscriptionsController < Api::V2::ApiController
     include Katello::Concerns::FilteredAutoCompleteSearch
 
     before_filter :find_activation_key
-    before_filter :find_system
+    before_filter :find_host
```

Comment:
Does every action need to find_host ? Same question for find_activation_key if you want to do some more cleanup :)